src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/OnOffClientFragment.kt

```diff
@@ -84,6 +89,39 @@ class OnOffClientFragment : Fragment() {
     }
   }
 
+  private fun updateAddressClick() {
+    val serviceInfo = NsdServiceInfo().apply {
+      serviceName = "%016X-%016X".format(fabricIdEd.text.toString().toLong(), deviceIdEd.text.toString().toLong())
+      serviceType = "_chip._tcp"
+    }
+
+    val resolverListener = object : NsdManager.ResolveListener {
+      override fun onResolveFailed(serviceInfo: NsdServiceInfo?, errorCode: Int) {
+        showMessage("Address resolution failed: $errorCode")
+      }
+
+      override fun onServiceResolved(serviceInfo: NsdServiceInfo?) {
+        val hostAddress = serviceInfo?.host?.hostAddress ?: ""
+        val port = serviceInfo?.port ?: 0
+
+        showMessage("Address: ${hostAddress}:${port}")
+
+        if (hostAddress == "" || port == 0)
+          return
+
+        try {
+          deviceController.updateAddress(deviceIdEd.text.toString().toLong(), hostAddress, port)
```

Comment:
This reverses the flow of lookup from what is expected by the spec. Why is the address looked-up by button click, rather than by internal operational discovery? The method here can only work temporarily, as long as the address of the peer doesn't change. It also does not account for commissionable discovery, which has a different format. 

The lookup of addresses is intended to be automatic over time, not manual based on user interaction.

This code may work in the short term, but it is baking many assumptions at the controller app which likely would need to be within the stack itself.

Reply:
Yes, I admitted all that in the description. There's currently no common code for triggering operational discovery, nor Android platform in CHIP code, but I need a temporary way to test operational discovery of Thread nodes with a proper border router setup. All existing controller apps call `ResolveNodeId` manually, so Android version is no different with regards to that.